lib/classes/record/adapter.php

```diff
@@ -1338,6 +1338,35 @@ public function insertTechnicalDatas(MediaVorus $mediavorus)
         return $this;
     }
 
+    /**
+     * Insert or update technical data
+     * $technicalDatas an array of name => value
+     *
+     * @param array $technicalDatas
+     */
+    public function insertOrUpdateTechnicalDatas($technicalDatas)
+    {
+        $technicalFields = media_subdef::getTechnicalFieldsList();
+        $sqlValues = null;
+
+        foreach($technicalDatas as $name => $value){
+            if(array_key_exists($name, $technicalFields)){
+                if(is_null($value)){
+                    $value = 0;
+                }
+                $sqlValues[] = [$this->getRecordId(), $name, $value, $value];
+            }
+        }
+
+        if($sqlValues){
+            $connection = $this->getDataboxConnection();
+            $connection->transactional(function (Connection $connection) use ($sqlValues) {
+                $statement = $connection->prepare('INSERT INTO technical_datas (record_id, name, value) VALUES (?, ?, ?) ON DUPLICATE KEY UPDATE value = ?');
+                array_walk($sqlValues, [$statement, 'execute']);
+            });
+        }
```

Comment:
ajouter $this->delete_data_from_cache(self::CACHE_TECHNICAL_DATA);

Reply:
ok